src/test/java/pl/umk/workshop/javaintroduction/StreamsSpec.java

```diff
@@ -0,0 +1,146 @@
+package pl.umk.workshop.javaintroduction;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+// Korzystając ze streamów javowych spraw aby testy przeszły
+@SpringBootTest
+public class StreamsSpec {
+
+    @Test
+        // Stwórz nową listę, która nie zawiera trzech pierwszych i ostatnich elementów
+    void cutting() {
+        //given
+        var elements = asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+
+        //when
+        var result = elements;
+
+        //then
+        assertEquals(result, asList(4, 5, 6, 7));
+    }
+
+    @Test
+        // Zmien pierwszą litere każdego imienia na wielką
+    void uppercase() {
+        //given
+        var names = asList("amadeusz", "jakub", "patryk", "mateusz", "dominik");
+
+        //when
+        var result = names;
+
+        //then
+        assertEquals(result, asList("Amadeusz", "Jakub", "Patryk", "Mateusz", "Dominik"));
+    }
+
+    @Test
+        // Utwórz jedną listę z list z sekcji 'given'. W połączonej liście elementy nie powinny się powtarzać.
+    void concatenation() {
+        //given
+        var firstList = asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+        var secondList = asList(2, 4, 6, 8, 10, 12, 14, 16, 18, 20);
+
+        //when
+        var result = emptyList();
+
+        //then
+        assertEquals(result, asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20));
+    }
+
+    @Test
+        // Wybierz imiona, których długość jest nieparzysta oraz posortuj je od najkrótszego do najdłużeszego
+    void nameSorting() {
+        //given
+        var names = asList("Amadeusz", "Mateusz", "Jakub", "Patryk", "Dominik");
+
+        //when
+        var result = names;
+
+        //then
+        assertEquals(result, asList("Jakub", "Mateusz", "Dominik"));
+    }
+
+    @Test
+        // Pogrupuj imiona ze względu na pierwszą literę.
+        // Oczekiwany output to Mapa gdzie kluczem jest litera a wartością imię
+        // Tip: Sprawdź Collectors
+    void groupingNames() {
+        //given
+        var names = asList("Monika", "Jakub", "Mateusz", "Dorota", "Julia", "Mieszko");
```

Comment:
Zamiast asList można też robić List.of

Reply:
Zostawiłem już w tej wersji na potrzeby zadań jest enough